lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -351,30 +348,34 @@ private SignedJWT validateHmrcInheritedIdentity(
         }
 
         // Validate VC in claim is valid
-        var signedInheritedIdentityJWT = SignedJWT.parse(inheritedIdentityJwtList.get(0));
+        var inheritedIdentityJwt = SignedJWT.parse(inheritedIdentityJwtList.get(0));
         var inheritedIdentityCriConfig = configService.getCriConfig(HMRC_MIGRATION_CRI);
 
         // The HMRC inherited identity VC will contain an HMRC-specific pairwise identifier
         // rather than our internal user id, so we cannot validate it against the OAuth user id.
         // Instead, SPOT will validate this when generating an identity bundle.
```

Comment:
This logic isn't quite going to work now - you're baking the `userId` into the `VerifiableCredential` that you build, but now it's not the right one.

I think we could do one (or more) of:
- Remove `userId` from `VerifiableCredential`, so it's not an issue
- Remove the special casing here, and pass a more explicit flag, e.g. `skipSubjectCheck`
  - This feels like the 'right' solution, since the current approach is obviously a hack (that I introduced...)
- Move this logic inside `VerifiableCredentialJwtValidator`, so that it knows about the inherited id special case
  - I don't like this!

Reply:
I think the second solution is the nicest! I'll go with that